src/plugins/vis_type_vega/public/data_model/search_api.ts

```diff
@@ -6,19 +6,38 @@
  * Side Public License, v 1.
  */
 
-import { combineLatest } from 'rxjs';
-import { map, tap } from 'rxjs/operators';
+import { combineLatest, from } from 'rxjs';
+import { map, tap, switchMap } from 'rxjs/operators';
 import { CoreStart, IUiSettingsClient } from 'kibana/public';
+import type { RuntimeFields } from '@elastic/elasticsearch/api/types';
+import { getData } from '../services';
 import {
   getSearchParamsFromRequest,
   SearchRequest,
   DataPublicPluginStart,
   IEsSearchResponse,
+  IndexPattern,
 } from '../../../data/public';
 import { search as dataPluginSearch } from '../../../data/public';
+import { ISearchRequestParams } from '../../../data/common';
 import { VegaInspectorAdapters } from '../vega_inspector';
 import { RequestResponder } from '../../../inspector/public';
 
+const extendSearchParamsWithRuntimeFields = (requestParams: ISearchRequestParams) =>
+  new Promise(async (resolve) => {
+    const indexPatternTitle = requestParams.index as string;
```

Comment:
why we need that casting to string?  if `requestParams.index` is undefined I don't see any reasons to execute code below